### PR TITLE
feat(sdk-core): Allow getting a staking wallet for any coin

### PR DESCRIPTION
**Pull request by mcalcano-bg**
allow getting a staking wallet for any coin

allow clients to send arbitrary properties when staking or unstaking

Ticket: BOS-649

### CODE REVIEW

File: modules/sdk-core/src/bitgo/staking/iStakingWallet.ts [hunk 1]
```diff
@@ -21,12 +21,16 @@ export interface StakingRequest {
 export interface StakeOptions {
   amount: string;
   clientId?: string;
+  validator?: string;
+  durationSeconds?: string;
+  [otherOptions: string]: unknown;
```
Comment (bitgoAaron) [inline modules/sdk-core/src/bitgo/staking/iStakingWallet.ts:26]: this kind of breaks the strict typing we prefer to use. Can you elaborate on the use case here?

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 2]
```diff
@@ -2104,9 +2104,6 @@ export class Wallet implements IWallet {
    * Create a staking wallet from this wallet
    */
   toStakingWallet(): StakingWallet {
-    if (!this.baseCoin.supportsStaking()) {
```
Comment (bitgoAaron) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:2107]: this method is no longer used anywhere, please also remove from baseCoin & all the other coins that override it.

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
needs clarification on `otherOptions` and cleanup dead code.

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 2 above)
Comment (mcalcano-bg) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:2107]: Fixed.

**Review by joey-bai — APPROVED** [review]

**Review by olibeyene — APPROVED** [review]

**Pull request merged**